devito/ir/clusters/cluster.py

```diff
@@ -193,6 +193,7 @@ def functions(self):
     def grid(self):
         grids = set(f.grid for f in self.functions if f.is_AbstractFunction)
         grids.discard(None)
+        grids = {g.root for g in grids}
```

Comment:
This will fail if there is no grid

Reply:
How? `grids` will be an empty set, thus triggering the first conditional
```python
>>> {g.root for g in set()}
set()
```